+replab/Centralvar.m

```diff
@@ -0,0 +1,232 @@
+            assert(self.dim == size(self.U,1), ['dimension is ', num2str(self.dim), ' but U is of size ', num2str(size(self.U,1)), 'x', num2str(size(self.U,2))]);
+            assert(self.dim == size(self.U,2), ['dimension is ', num2str(self.dim), ' but U is of size ', num2str(size(self.U,1)), 'x', num2str(size(self.U,2))]);
+
+            % We construct the SDP matrix by block
+            self.blocks = cell(1,self.nComponents);
+            for i = 1:self.nComponents
+                switch self.types(i)
+                    case 'R'
+                        self.blocks{i} = sdpvar(self.multiplicities(i));
+                    case 'C'
+                        assert(self.dimensions1(i) == 2, ['Dimension ', num2str(self.dimensions1(i)), ' for a complex representation is currently not supported.']);
+                        self.blocks{i} = kron(sdpvar(self.multiplicities(i)), eye(2));
+                        tmp = kron(sdpvar(self.multiplicities(i), self.multiplicities(i), 'skew'), [1 0; 0 -1]);
+                        reOrder = reshape(1:2*self.multiplicities(i), 2, self.multiplicities(i));
+                        reOrder = reOrder([2 1],:);
+                        reOrder = reOrder(:)';
+                        self.blocks{i} = self.blocks{i} + tmp(:, reOrder);
+                    case 'H'
+                        assert(self.dimensions1(i) == 4, ['Dimension ', num2str(self.dimensions1(i)), ' for a quaternionic representation is currently not supported.']);
+                        da = replab.DivisionAlgebra.quaternion;
+                        self.blocks{i} = sdpvar(1)*ones(4*self.multiplicities(i));
+                        for x = 1:self.multiplicities(i)
+                            for y = x:self.multiplicities(i)
+                                vars = sdpvar(4,1);
+                                if (x == y)
+                                    % Only keep the symmetric part
+                                    self.blocks{i}(4*(x-1)+[1:4], 4*(y-1)+[1:4]) = da.toMatrix([1 0 0 0]'.*vars);
+                                else
+                                    % keep everything
+                                    self.blocks{i}(4*(x-1)+[1:4], 4*(y-1)+[1:4]) = da.toMatrix(vars);
+                                    % Set the off-diagonal terms as well
+                                    self.blocks{i}(4*(y-1)+[1:4], 4*(x-1)+[1:4]) = da.toMatrix([1 -1 -1 -1]'.*vars);
+                                end
+                            end
+                        end
+                    otherwise
+                        error('Unknown type');
+                end
+                
+                % sanity check
+                assert(issymmetric(self.blocks{i}));
+            end
+
+        end
+
+    end
+    
+    methods (Static) % Factory methods
+        
+        function R = fromGenerators(generators)
```

Comment:
I'm not sure in what way the name ambiguous: what exactly could a user wrongly interpret here?

The aim of this method is not to be most general. I am also looking forward to add other constructors of the kind of `fromRepresentation`. The main purpose of the current method is to provide a first simple constructor which requires minimum knowledge of the underlying mechanics to instantiate a `CentralVar` object.

The problem I see with the name `fromSignedPermutations` is that in full logic its arguments should then be an object of type `SignedPermutations`. Should this then not just be another possible constructor of its own? I tried to use here as input type what is used elsewhere in the code to describe generators, namely a cell array of vectors.

Reply:
Actually, signed permutations are just described by (row) vectors of integers describing the images of 1..n; the `SignedPermutations` class describes the *group* of signed permutations, and specifies how they compose, etc..

Do you think there would be other objections to the name `fromSignedPermutations`?

(Future updates to `RepLAB` will de-emphasize the role of signed permutations as primitives, as we introduce group products, etc..)

Reply:
Hmm, ok I think I start to understand what you mean. So what you are saying is that we should not expect a method called `fromSignedPermutations` to accept as an argument an object of type `SignedPermutations`, because this class does not actually represent a signed permutation?! Isn't this misleading?

Honestly, I think a method called `fromSomething` should accept as argument objects of type `Something`. Please tell me either how you want me to rename the class `SignedPermutations`, or if you prefer to keep the method momentarily named `fromGenerators` until changes that should happen on that part of the code are stabilized, or if you have a better suggestion!-)